### ISSUE TITLE
[BUG] Nurikabe Black Bottle Neck Rule Lacks Testing

### ISSUE THREAD

**Issue by walkingjason** [Bram-Hub/LEGUP#435]
### Description

The only test for Black Bottle Neck Rule is commented out: https://github.com/Bram-Hub/LEGUP/blob/master/src/test/java/puzzles/nurikabe/rules/BlackBottleNeckBasicRuleTest.java#L31

Uncommenting this test results in an error.

### Steps to Reproduce

1. Uncomment https://github.com/Bram-Hub/LEGUP/blob/master/src/test/java/puzzles/nurikabe/rules/BlackBottleNeckBasicRuleTest.java#L31
2. Compile. 

### Expected Behavior

No error in compile time.

### Screenshots

![image](https://user-images.githubusercontent.com/29169987/217378049-661fc466-1856-42e6-8622-95672ca611fc.png)

### Additional Context

Attempt to test multiple grid changes only after this [issue ](https://github.com/Bram-Hub/LEGUP/issues/95)is resolved.

**Comment by walkingjason** [comment]
@Avilez-dev-11 A branch [commit ](https://github.com/Bram-Hub/LEGUP/pull/469/commits/9527950084d6f76a21a614804f97cd14c061c59f)has been merged into dev which changes quite a bit of the logic used in WhiteBottleNeckRule. [This issue](https://github.com/Bram-Hub/LEGUP/issues/95) still exists on BlackBottleNeckRule, and I believe the new logic for WhiteBottleNeckRule helps. You can also let me take over if you feel not much substantial progress has been made.

**Comment by Avilez-dev-11** [comment]
@walkingjason You can feel free to take over the issue as I don't feel much substantial progress has been made, and have moved to a different issue.

**Comment by walkingjason** [comment]
This issue has become a part of another [issue](https://github.com/Bram-Hub/LEGUP/issues/510) that I will be working on. Closing this issue to avoid redundancy.
